lib/avromatic/model/serialization.rb

```diff
@@ -41,7 +41,7 @@ def avro_hash(fields)
             result[key.to_s] = if value.is_a?(Avromatic::Model::Attributes)
                                  value.value_attributes_for_avro
                                else
-                                 value
+                                 avro_coder[key].try(:call, value) || value
```

Comment:
The fullname of the type is not readily accessible at this point. We could prebuild a map from the field names to the Avro fields. The we could lookup name -> Avro field -> custom type.

Here, I just prebuilt the map from field name to coder.